src/sqlfluff/rules/L059.py

```diff
@@ -76,4 +158,26 @@ def _eval(self, context: RuleContext) -> Optional[LintResult]:
                 description=f"Unnecessary quoted identifier {context.segment.raw}.",
             )
 
+        # If this is a reserved keyword, or it is not a valid naked identifier,
+        # ensure the quote used is the preferred one.
+        expected = f"{quote}{quoted_identifier_contents}{quote}"
+        if expected != context.segment.raw:
+            # If the identifier contents contains any occurrence of the preferred quote,
+            # ignore this rule for now.
+            if quote in quoted_identifier_contents:
+                # TODO: if different levels of criticity can be set, this case should
+                #       trigger an `info` level.
+                return None
```

Comment:
Could this be moved way up to the top to exit early?

Reply:
Actually, on those lines, I know I'm in a context where I'm dealing with reserved keywords or identifiers with special chars. If I move this way up, it would means I also extend the context to any kind of identifier, and moving it will change the semantic.

Reply:
But if an identifier contains the quote you want to use, you'll never change it will you? So don't see why you can't check that early and drop out?

Reply:
Yep, definitely right.
Updated.